File: virtualization-lb/src/main/java/co/edu/escuelaing/lb/repositories/MessageRepository.java
package co.edu.escuelaing.lb.repositories;

import co.edu.escuelaing.lb.models.Message;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

import java.util.List;

public interface MessageRepository {

    @GET("messages")
    Call<List<Message>> getMessages();

    @POST("messages")
    Call<Message> createMessage(@Body Message message);
}

File: virtualization-front/src/repositories/MessageRepository.js
const axios = require('axios');

const configuration = {
    headers: {
        "Access-Control-Allow-Origin": "*"
    }
}

const MessageRepository = {}

MessageRepository.getMessages = function(){
    
    return axios.get('/messages', configuration).then(response => response.data)
}

MessageRepository.createMessage = function(message){
    
    return axios.post('/messages', {
        message
    },configuration).then(response => response.data)
}

export default MessageRepository;